cachito/web/content_manifest.py

```diff
@@ -232,3 +237,216 @@ def generate_icm(self, image_contents=None):
         icm = deepcopy(BASE_ICM)
         icm["image_contents"] = image_contents or []
         return deep_sort_icm(icm)
+
+
+class Package:
+    """
+    A package within a content manifest.
+
+    It is used primarily to generate a package URL (purl).
+    """
+
+    def __init__(self, name, type, version, dev=False, dependencies=None, path=None, replaces=None):
+        """
+        Initialize ContentManifestPackage.
+
+        :param str name: the package's name
+        :param str type: the package manager used to process this package. Should be one
+            of the following: gomod, go-package, npm, yarn, pip or git-submodule
+        :param str version: the package's version
+        :param bool dev: if the package is only used in a dev environment
+        :param list[dict] dependencies: the package's dependencies as a list of JSON
+            dicts
+        :param str path: relative path to package from root of repository
+        :param str replaces: the dependency that is being replaced by this one in the
+            context of the request
+        """
+        self.name = name
+        self.type = type
+        self.version = version
+        self.dev = dev
+        self.path = path
+        self.replaces = replaces
+
+        if not dependencies:
+            dependencies = []
+
+        self.dependencies = [Package.from_json(dependency) for dependency in dependencies]
+
+    def __repr__(self):
+        return (
+            f"<{self.__class__.__name__} name={self.name}, type={self.type}, "
+            f"version={self.version}>"
+        )
+
+    def __hash__(self):
+        return hash((self.name, self.type, self.version, self.dev))
+
+    def __eq__(self, o: object):
+        if type(object) != self.__class__:
+            return False
```

Comment:
```suggestion
        if not isinstance(o, Package):
            return False
```
The typo of o -> object would also make this always `False`, it may be good to have a unit test for package equality